yarn_api_client/base.py

```diff
@@ -1,58 +1,72 @@
 # -*- coding: utf-8 -*-
 from __future__ import unicode_literals
-try:
-    from httplib import HTTPConnection, OK
-except ImportError:
-    from http.client import HTTPConnection, OK
+
 import json
 import logging
-try:
-    from urllib import urlencode
-except ImportError:
-    from urllib.parse import urlencode
+import requests
 
+from requests_kerberos import HTTPKerberosAuth
 from .errors import APIError, ConfigurationError
 
 
 class Response(object):
-    def __init__(self, http_response):
-        self.data = json.load(http_response)
+    def __init__(self, response):
+        self.data = response.json()
 
 
 class BaseYarnAPI(object):
     response_class = Response
 
+    def _validateConfiguration(self):
+        if self.address is None:
+            raise ConfigurationError('API address is not set')
+        elif self.port is None:
+            raise ConfigurationError('API port is not set')
+
     def request(self, api_path, **query_args):
-        params = urlencode(query_args)
-        if params:
-            path = api_path + '?' + params
+        params = query_args
+        api_endpoint = 'http://{}:{}{}'.format(self.address, self.port, api_path)
```

Comment:
I have a doubt about hardcoding `http` here

Reply:
enabling https requires some more work throughout the code, we could address that in the future